algorithms/src/std_algorithms/impl/Kokkos_Search.hpp

```diff
@@ -149,13 +153,99 @@ IteratorType1 search_impl(const std::string& label, const ExecutionSpace& ex,
 }
 
 template <class ExecutionSpace, class IteratorType1, class IteratorType2>
-IteratorType1 search_impl(const std::string& label, const ExecutionSpace& ex,
-                          IteratorType1 first, IteratorType1 last,
-                          IteratorType2 s_first, IteratorType2 s_last) {
+IteratorType1 search_exespace_impl(const std::string& label,
+                                   const ExecutionSpace& ex,
+                                   IteratorType1 first, IteratorType1 last,
+                                   IteratorType2 s_first,
+                                   IteratorType2 s_last) {
+  using value_type1    = typename IteratorType1::value_type;
+  using value_type2    = typename IteratorType2::value_type;
+  using predicate_type = StdAlgoEqualBinaryPredicate<value_type1, value_type2>;
+  return search_exespace_impl(label, ex, first, last, s_first, s_last,
+                              predicate_type());
+}
+
+//
+// team impl
+//
+template <class TeamHandleType, class IteratorType1, class IteratorType2,
+          class BinaryPredicateType>
+KOKKOS_FUNCTION IteratorType1
+search_team_impl(const TeamHandleType& teamHandle, IteratorType1 first,
+                 IteratorType1 last, IteratorType2 s_first,
+                 IteratorType2 s_last, const BinaryPredicateType& pred) {
+  // checks
+  Impl::static_assert_random_access_and_accessible(teamHandle, first, s_first);
+  Impl::static_assert_iterators_have_matching_difference_type(first, s_first);
+  Impl::expect_valid_range(first, last);
+  Impl::expect_valid_range(s_first, s_last);
+
+  // the target sequence should not be larger than the range [first, last)
+  namespace KE                        = ::Kokkos::Experimental;
+  const auto num_elements             = KE::distance(first, last);
+  [[maybe_unused]] const auto s_count = KE::distance(s_first, s_last);
```

Comment:
Why is this unused? Which compiler complains?

Reply:
the complain comes from line 189 and 193, it sees this possibility of existing earlier. I don't remember which compiler is doing this but definitely was getting warnings. I can dig it back up if you really want to know

Reply:
the same should be true for num_elements though